### PR TITLE
feat: title, subject, key-stage, topic agents, and error handling [AI-1217]

### PR DESCRIPTION
**Pull request by mantagen**
## Description

This PR adds user control over core lesson plan fields and implements robust error handling for schema validation:

### New User Controls
- **Basic field editing**: Users can now directly manipulate title, keyStage, subject, topic, and additionalMaterials sections
- **Complete section coverage**: All lesson plan sections now have corresponding agents

### Robust Schema Validation & Error Handling
- **Schema enforcement**: Implemented strict validation to prevent users from breaking the system by requesting invalid content (e.g., too many keywords/misconceptions). We use a loose schema for structured output (as LLMs can't handle e.g. max/min lengths), with a strict parse/retry system.
- **Retry logic**: When agents fail schema validation, the system automatically retries with corrected instructions
- **Clear user feedback**: Users receive helpful error messages when Aila can't complete requests due to schema constraints, rather than system crashes

### Fixed RAG System
- **Reliable lesson retrieval**: Fixed relevant lessons (RAG) functionality with proper error handling for malformed data
- **Content anchoring**: RAG system now works properly, though content injection for "based on" lessons still needs work (see [ticket](https://www.notion.so/oaknationalacademy/Add-prompt-detail-for-based-on-content-ensure-it-s-properly-injected-1fc26cc4e1b18041a03bd8714a9cfa40?pvs=4))

### 1. User Field Manipulation
- Edit title, keyStage, subject, topic, and additionalMaterials sections individually
- Verify changes are applied correctly and system remains stable

### 2. Schema Validation Robustness  
- **Test breaking requests**: Try asking for more keywords/misconceptions than allowed
- **Verify error handling**: Confirm system doesn't crash and provides clear feedback like "I can only create up to X keywords"
- **Test edge cases**: Try other schema-violating requests (invalid formats, wrong data types)

The key improvement is that users shouldn't be able to break Aila any more - it will handle invalid requests gracefully and explain why it can't complete them.

## Schema Validation Constraints

Part of this PR ensures that the limits/constraints included in the prompts are enforced at the code level (in the Agentic flow, i.e. when the feature flag is on)

The following sections now have strict validation that prevents users from generating content beyond specified limits:

- **Learning Outcome**: Limited to maximum 190 characters to ensure proper display in UI components
- **Learning Cycles**: Limited to maximum 3 cycles  
- **Prior Knowledge**: Limited to maximum 5 points 
- **Key Learning Points**: Limited to maximum 5 points
- **Misconceptions**: Limited to max 3 
- **Keywords**: Limited to max 5
- **Starter Quiz**: Limited to maximum 6 questions to fit typical lesson timeframes
- **Exit Quiz**: Limited to maximum 6 questions to fit typical lesson timeframes  
- **Learning Cycles Content**: Check for understanding has minimum 2 questions

When users request content that exceeds these limits, the system will automatically retry generation within the constraints and provide clear feedback about the limitations rather than failing or crashing.

### CODE REVIEW

File: packages/aila/src/lib/agents/createOpenAIClient.ts [hunk 1]
```diff
@@ -1,9 +0,0 @@
-import OpenAI, { type ClientOptions } from "openai";
```
Comment (mantagen) [inline packages/aila/src/lib/agents/createOpenAIClient.ts:1]: File was unused

File: packages/aila/src/lib/agents/promptAgentHandler.ts [hunk 2]
```diff
@@ -19,54 +25,150 @@ export async function promptAgentHandler<Schema extends SchemaWithValue>({
...
+  const responseFormat = zodTextFormat(schema, `${agent.name}_response_schema`);
 
-  const result = await openAIClient.responses.parse({
-    instructions: agent.prompt,
-    input: `### Examples from similar lessons
+  const instructions = agent.prompt;
+  const input = `### Examples from similar lessons
 
-${ragData.map((lesson) => `#### ${targetKey} ${lesson.title}\n\n${agent.extractRagData(lesson)}`).join("\n\n")}
+${ragData.map((lesson, i) => `#### Example ${i + 1}: ${targetKey} from lesson '${lesson.title}'\n\n${agent.extractRagData(lesson)}`).join("\n\n")}
```
Comment (mantagen) [inline packages/aila/src/lib/agents/promptAgentHandler.ts:57]: Will add some more colour to this in [follow up work](https://www.notion.so/oaknationalacademy/Add-prompt-detail-for-based-on-content-ensure-it-s-properly-injected-and-improve-on-relevant-le-1fc26cc4e1b18041a03bd8714a9cfa40?source=copy_link)

File: packages/aila/src/lib/agents/promptAgentHandler.ts [hunk 3]
```diff
@@ -19,54 +25,150 @@ export async function promptAgentHandler<Schema extends SchemaWithValue>({
...
+       * The reason we need this is that structured output does not support
+       * all the schema validation that we need.
+       */
+      const inputWithRetryInstructions =
+        input +
+        `\n\n### Note\nThis is a retry because the previous response was not valid. Your previous response output was: ${JSON.stringify(
+          result.output_parsed.output.value,
+        )}, however the strict schema is: ${JSON.stringify(
+          agent.schemaStrict,
+        )}; the error was ${JSON.stringify(strictParseResult.error.errors)} If the user's request is at odds with the schema, please respond with a message explaining the issue. Another agent will handle messaging to the user.`;
```
Comment (codeincontext) [inline packages/aila/src/lib/agents/promptAgentHandler.ts:98]:
```suggestion
        )}; the error was ${JSON.stringify(strictParseResult.error.errors)} If the user's request is at odds with the schema, please respond with an internal message explaining the issue. Another agent will handle messaging to the user.`;
```

Could this help clarify the message it's making?

File: packages/aila/src/lib/agents/prompts/messageToUserInstructions.ts [hunk 4]
```diff
@@ -48,10 +48,11 @@ The message history includes prior dialogue and shows that Aila (powered by a mu
...
-1. \`basedOn\`, \`learningOutcome\`, \`learningCycles\`  
-2. \`priorKnowledge\`, \`keyLearningPoints\`, \`misconceptions\`, \`keywords\`  
-3. \`starterQuiz\`, \`cycle1\`, \`cycle2\`, \`cycle3\`, \`exitQuiz\`  
-4. \`additionalMaterials\`
+# Section Groups (by default processed sequentially):  
+1. title, keyStage, subject, topic
+1. basedOn, learningOutcome, learningCycles
+1. priorKnowledge, keyLearningPoints, misconceptions, keywords
+1. starterQuiz, cycle1, cycle2, cycle3, exitQuiz
+1. additionalMaterials
```
Comment (codeincontext) [inline packages/aila/src/lib/agents/prompts/messageToUserInstructions.ts:56]: Should these all be `1.`?

**Review by codeincontext — COMMENTED** [review]
I don't have a lot of context on how the agentic approach works, but looking at the diff here it seems to make sense. 👍

On code style: promptAgentHandler has lengthier if/else statements than I personally prefer, but it's not important at all at this stage

File: packages/aila/src/lib/agents/prompts/routerInstructions.ts [hunk 5]
```diff
@@ -2,11 +2,12 @@ export const routerInstructions = `You are a planning agent that supports users
...
-### Section Groups (process sequentially):
+### Section Groups (by default processed sequentially):
+1. title, keyStage, subject, topic
 1. basedOn, learningOutcome, learningCycles
-2. priorKnowledge, keyLearningPoints, misconceptions, keywords
-3. starterQuiz, cycle1, cycle2, cycle3, exitQuiz
-4. additionalMaterials
+1. priorKnowledge, keyLearningPoints, misconceptions, keywords
+1. starterQuiz, cycle1, cycle2, cycle3, exitQuiz
+1. additionalMaterials
```
[the remark this answers is not in the document]
Comment (mantagen) [reply packages/aila/src/lib/agents/prompts/routerInstructions.ts:10]: Yeah fair point. I had them as `1.` for maintainability, and many markdown parsers will automatically assign the correct number if you give them all number 1, so I guess assumed the LLM would naturally handle it similarly.

**Review by mikeritson-oak — APPROVED** [review]
Each section has been checked individually to confirm that they adhere to the rules defined in this PR. Sections appeared with content within the rules and could not be maniupulated to break the rules. We have spotted two minor issue which will need addressing at a later point:

1. Chat output messages sometimes confirm that changes have been made which haven't (due to the section rules)
2. Bullet points appear doubled up in the learning cycles
